models/classes/QtiResultsService.php

```diff
@@ -354,4 +355,23 @@ private function getDisplayDate(string $epoch): string
     {
         return tao_helpers_Date::displayeDate($epoch, tao_helpers_Date::FORMAT_ISO8601);
     }
+
+
+    private function prepareItemVariableValue($value, $basetype): string
+    {
+        if ($basetype === 'file') {
+
+            if (extension_loaded('fileinfo')) {
+                $info = new finfo(FILEINFO_MIME_TYPE);
+                $mimeType = $info->buffer($value);
+            } else {
+                // fallback
+                $mimeType = 'application/octet-stream';
+            }
+
+            return sprintf('%s,base64,%s', $mimeType, base64_encode($value));
```

Comment:
Could we transform this as an external "helper" method so that we can unit test it? It might be also reused in the future to have an homogeneous treatment elsewhere.

Reply:
This might be part of the SDK indeed... ;-)

Reply:
Done